### PR TITLE
add server call

### PR DESCRIPTION
**Pull request by BusyJay**
This pr adds functions needed to handle rpc on server side.

### CODE REVIEW

File: compiler/src/codegen.rs [hunk 1]
```diff
@@ -189,7 +189,7 @@ impl<'a> MethodGen<'a> {
     fn duplex_streaming(&self, method_name: &str) -> String {
         format!("{}(&self) -> {}<{}, {}>",
                 method_name,
-                fq_grpc("DuplexStreamingCallHandler"),
+                fq_grpc("DuplexCallHandler"),
```
Comment (siddontang) [inline compiler/src/codegen.rs]: why remove Streaming here?

File: compiler/src/codegen.rs [hunk 2]
```diff
@@ -189,7 +189,7 @@ impl<'a> MethodGen<'a> {
     fn duplex_streaming(&self, method_name: &str) -> String {
```
Comment (siddontang) [inline compiler/src/codegen.rs]: please use another PR for the compiler.

File: compiler/src/codegen.rs (hunk 1 above)
Comment (BusyJay) [inline compiler/src/codegen.rs]: The name is too long.

File: compiler/src/codegen.rs (hunk 2 above)
Comment (BusyJay) [inline compiler/src/codegen.rs]: No, they are related. Not changing the compiler will break master.

File: src/call/mod.rs [hunk 3]
```diff
@@ -232,6 +235,40 @@ impl Call {
...
+                                                            ptr::null(),
+                                                            0,
+                                                            0,
+                                                            prom_ptr as *mut c_void)
+        };
+        if code != GrpcCallStatus::Ok {
+            unsafe {
+                Box::from_raw(prom_ptr);
+            }
+            panic!("create call fail: {:?}", code);
```
Comment (siddontang) [inline src/call/mod.rs]: can we meet error here?

Comment (BusyJay) [inline src/call/mod.rs]: Yes. If it happens, it means the library is implemented incorrectly.

File: src/call/mod.rs [hunk 4]
```diff
@@ -232,6 +235,40 @@ impl Call {
...
+
+        let code = unsafe {
+            let details_ptr = status
+                .details
+                .as_ref()
+                .map_or_else(ptr::null, |s| s.as_ptr() as _);
+            let details_len = status.details.as_ref().map_or(0, String::len);
+            grpc_sys::grpcwrap_call_send_status_from_server(call_ptr,
+                                                            batch_ptr,
+                                                            GrpcStatusCode::Unimplemented,
```
Comment (disksing) [inline src/call/mod.rs]: status.status

File: src/call/mod.rs [hunk 5]
```diff
@@ -232,6 +235,40 @@ impl Call {
         })
     }
 
+    /// Abort a rpc call before handler is called.
```
Comment (disksing) [inline src/call/mod.rs]: an rpc call

**Comment by disksing** [comment]
LGTM.

**Pull request merged**